embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/session/SessionHandler.kt

```diff
@@ -77,6 +77,11 @@ internal class SessionHandler(
     @VisibleForTesting
     var scheduledFuture: ScheduledFuture<*>? = null
 
+    /**
+     * Guards session state changes.
+     */
+    private val lock = Any()
```

Comment:
If we're not ensuring that start/stop are done atomically, we are not preventing the case where data can be recorded while there's no active session? It was probably a problem before, but is there a plan to solve that later?